[Feature][Deploy] Add missing default values in dolphinscheduler_env.sh

EricGao888 opened the pull request:
## Purpose of the pull request
* Add missing default values in dolphinscheduler_env.sh
* Update related docs
* This PR closes: #9675 

## Brief change log
* Already Described above

## Verify this pull request
* Verified by manual tests

EricGao888 committed:
[Feature][Deploy] Add missing default values in dolphinscheduler_env.sh (#9675)

SbloodyS commented:
Hi @EricGao888 , please resolve conficts.

EricGao888 commented:
> Hi @EricGao888 , please resolve conficts.

@SbloodyS Thx for the reminder! I'm a bit confused about what to do in this PR, since we no more need add default values to `dolphinscheduler_env.sh` as explained in this comment: https://github.com/apache/dolphinscheduler/issues/9675#issuecomment-1107996038

SbloodyS commented:
@EricGao888 Finally, we have an agreement. please see  https://github.com/apache/dolphinscheduler/issues/9675#issuecomment-1108041268

EricGao888 commented:
> @EricGao888 Finally, we have an agreement. please see [#9675 (comment)](https://github.com/apache/dolphinscheduler/issues/9675#issuecomment-1108041268)

@SbloodyS I just checked the latest code, it seems this feature has been implemented by PR #9726 
https://github.com/apache/dolphinscheduler/blob/637028735dbf8a78efa63cf8358d2bead7bb5c6e/script/dolphinscheduler-daemon.sh#L37-L51

SbloodyS commented:
Yes. But some of the default value in ```dolphinscheduler_env.sh``` still missing. @EricGao888 
https://github.com/apache/dolphinscheduler/blob/637028735dbf8a78efa63cf8358d2bead7bb5c6e/script/env/dolphinscheduler_env.sh#L24-L27

EricGao888 commented:
> Yes. But some of the default value in `dolphinscheduler_env.sh` still missing. @EricGao888
> 
> https://github.com/apache/dolphinscheduler/blob/637028735dbf8a78efa63cf8358d2bead7bb5c6e/script/env/dolphinscheduler_env.sh#L24-L27

Sure I will fix it.

EricGao888 commented:
I added missing default values in `dolphinscheduler_env.sh`, updated related docs and fixed minor issue in `update-schema.sh`  following PR #9726, PTAL when you have time. Thx! @zhongjiajie @SbloodyS 

BTW, not sure why some E2E tests failed.

SbloodyS commented:
```script/env/dolphinscheduler_env.sh``` will overwrite the default setting of stanalone server. Standalone server starts with ```H2``` database by default. That's why E2E tests failed. 

We may find a way bypass it. Otherwise, restore ```script/env/dolphinscheduler_env.sh``` as it is. @EricGao888

EricGao888 commented:
> `script/env/dolphinscheduler_env.sh` will overwrite the default setting of stanalone server. Standalone server starts with `H2` database by default. That's why E2E tests failed.
> 
> We may find a way bypass it. Otherwise, restore `script/env/dolphinscheduler_env.sh` as it is. @EricGao888

@SbloodyS I think it may not be easy to find a way to bypass it. Considering we already have default values in `application.yaml` and also provider users with examples in docs in this PR, should we just restore `script/env/dolphinscheduler_env.sh` and leave those default values blank? https://github.com/apache/dolphinscheduler/blob/52b188b3479dadf54d160bacfe5b19c3fd82d038/dolphinscheduler-tools/src/main/resources/application.yaml#L17-L24
![image](https://user-images.githubusercontent.com/34905992/165311988-e59b2ffa-719f-47b5-b3bc-c112421d7cab.png)

SbloodyS commented:
@EricGao888 I think these examples is enough for users.

EricGao888 committed:
[Feature][Deploy] Restore dolphinscheduler_env.sh to avoid e2e failure (#9675)

EricGao888 commented:
> @EricGao888 I think these examples is enough for users.

Done, CI looks good now. : )

SbloodyS commented on docs/docs/en/guide/installation/pseudo-cluster.md:
````diff
@@ -168,13 +171,20 @@ mysql> GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'%';
+```shell
+export DATABASE=${DATABASE:-mysql}
+export SPRING_PROFILES_ACTIVE=${DATABASE}
+export SPRING_DATASOURCE_DRIVER_CLASS_NAME=com.mysql.jdbc.Driver
````
The mysql package version used by ds is ```8.0.16```. Therefore we should use ```com.mysql.cj.jdbc.Driver``` instead of ```com.mysql.jdbc.Driver```

EricGao888 replied:
Fixed in the latest commit. Thx for the review : )

EricGao888 committed:
[Feature][Deploy] Change mysql driver class name to com.mysql.cj.jdbc.Driver (#9675)

SbloodyS reviewed (dismissed):
LGTM if CI passed. Do you have time to take a look? @zhongjiajie

zhongjiajie commented:
I am take a look on this PR now

zhongjiajie commented on docs/docs/en/guide/installation/pseudo-cluster.md:
```diff
@@ -115,26 +97,47 @@ deployUser="dolphinscheduler"
-export SPRING_DATASOURCE_USERNAME="username"
-export SPRING_DATASOURCE_PASSWORD="password"
+export SPRING_DATASOURCE_URL=jdbc:postgresql://127.0.0.1:5432/dolphinscheduler
+export SPRING_DATASOURCE_USERNAME={user}
+export SPRING_DATASOURCE_PASSWORD={password}
```
why we use `{user}` and `{password}` instead of exactly string value

EricGao888 replied:
I think it may be able to remind users of replacing them with their own. I can change it to string value.

zhongjiajie replied:
maybe you're right, keep it pls 👍

zhongjiajie commented on docs/docs/en/guide/installation/pseudo-cluster.md:
```diff
@@ -115,26 +97,47 @@ deployUser="dolphinscheduler"
-and some tasks which need external dependencies or libraries such as `JAVA_HOME` and `SPARK_HOME`. You could ignore the
-task external dependencies if you do not use those tasks, but you have to change `JAVA_HOME`, registry center and database
+File  `./bin/env/dolphinscheduler_env.sh` describes the following configurations:
+* Database configuration of DolphinScheduler, see [Initialize the Database](#initialize-the-database) for detailed instructions.
```
Add one more blank line will be more confortable
```suggestion

* Database configuration of DolphinScheduler, see [Initialize the Database](#initialize-the-database) for detailed instructions.
```

zhongjiajie replied:
Same as ZH docs

EricGao888 replied:
Sure I will fix it.

zhongjiajie commented on dolphinscheduler-tools/src/main/bin/create-schema.sh:
```diff
@@ -18,7 +18,8 @@
 DOLPHINSCHEDULER_HOME=${DOLPHINSCHEDULER_HOME:-$(cd $BIN_DIR/../..; pwd)}
-source "$DOLPHINSCHEDULER_HOME/tools/bin/dolphinscheduler_env.sh"
+
+source "$DOLPHINSCHEDULER_HOME/bin/env/dolphinscheduler_env.sh"
```
if we do not use `dolphinscheduler_env.sh` file anymore, we should also remove the assembly config https://github.com/apache/dolphinscheduler/blob/de50f43de63e42488cb38169cd56fba77dd24fc6/dolphinscheduler-tools/src/main/assembly/dolphinscheduler-tools.xml#L41-L49

EricGao888 replied:
I think it is a good idea to remove the assembly config in case users get confused. Will fix it in the next commit. : )

EricGao888 replied:
@SbloodyS @zhongjiajie E2E tests failed. May I ask is it caused by the removal of assembly config? Is there any way to bypass it? Thx!

zhongjiajie replied:
I do not think it will cause the failed. I will restart the CI

SbloodyS replied:
@EricGao888 It's E2E build failed. Maybe it's the problem with GitHub Actions since it can't guarantee 99.99% stability...

zhongjiajie replied:
> @EricGao888 It's E2E build failed. Maybe it's the problem with GitHub Actions since it can't guarantee 99.99% stability...

could I @github at this time？ 🤣

EricGao888 replied:
@github +1

EricGao888 committed:
[Feature][Deploy] Remove redundant config file (#9675)

zhongjiajie reviewed (approved):
LGTM, @SbloodyS do you have addition suggestion?

SbloodyS reviewed (approved):
+1
